### DISCUSSION TITLE
Change in plan for Weak BC

### DISCUSSION THREAD

**Discussion by Goldeninscriptions** [APSIS-ANALYSIS/PERIGEE#110] [Ideas]
1. Design new file names and new class names, ElemBC_3D_turbulent_wall etc.

2. Design a new basic class for ElemBC, with which we use grad(NA) of volume element to calculate the surface integral.

3. Move C_bI and Q from preprocessor to driver.

4. Design a set of  QuadPts_Gauss_..._boundary(boundary_id) classes

5. Design a member of FEAElement which connects to the boundary element, and a member function to get the outward normal and Jacobian at boundary quadrature points

**Comment by ju-liu** [comment]
我们可以考虑将face_id作为一个optional的member data放置到目前的elembc_3d中。这样可以兼顾之前的实现，同时对于需要遍历体单元的情况，我们可以通过face_id加global_cell获得三角/四边形对应的tet或hex单元的全部信息。
